### ISSUE TITLE
API bug blocked by cors

### ISSUE THREAD

**Issue by judicaelandria** [forem/forem#16751]
I am reporting a bug on the API that require API Key
it's always blocked by cors

like if we do

```js
fetch("https://dev.to/api/articles/me", {
        headers: {
          "api-key": "API_KEY",
        },
      })
        .then((res) => console.log({ res }))
        .catch((err) => console.log({ err }));
```

**Comment by djuber** [comment]
When I try from about:blank (where origin is null) I get a cors error indicating the api key header is not allowed:

```
Access to fetch at 'https://dev.to/api/articles/me' from origin 'null' has been blocked by CORS policy: Request header field api-key is not allowed by Access-Control-Allow-Headers in preflight response.
```

When I try from other pages (like this one) I get a content security policy error ["connect-src"](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/connect-src), but that may be unrelated to the issue with the disallowed header.

**Comment by djuber** [comment]
duplicate https://github.com/forem/forem/issues/7878  ?

**Comment by judicaelandria** [comment]
Yes I used the api that doesn't need the api key header but there is field that I want to get too but it's only available when using the api key, need to be authenticated
